File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/instruction/tree/InitialTreeInsnExecutor.java
package net.yakclient.mixin.base.internal.instruction.tree;

import net.yakclient.mixin.base.internal.instruction.Instruction;
import net.yakclient.mixin.base.internal.instruction.InstructionExecutor;
import org.objectweb.asm.tree.InsnList;

public class InitialTreeInsnExecutor implements InstructionExecutor {
    private final InsnList insnList;
    final Instruction insn;

    public InitialTreeInsnExecutor(InsnList insnList, Instruction insn) {
        this.insnList = insnList;
        this.insn = insn;
    }

    @Override
    public void execute() {
        this.insnList.insert(this.insn.getInstructions());
    }
}

File: base/src/main/kotlin/net/yakclient/mixin/base/internal/instruction/tree/InitialTreeInsnExecutor.kt
package net.yakclient.mixin.base.internal.instruction.tree

import net.yakclient.mixin.base.internal.instruction.Instruction
import net.yakclient.mixin.base.internal.instruction.InstructionExecutor
import org.objectweb.asm.tree.InsnList

class InitialTreeInsnExecutor(private val insnList: InsnList, val insn: Instruction) : InstructionExecutor {
    override fun execute() {
        insnList.insert(insn.insn)
    }
}